src/tools/illink/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -106,11 +106,21 @@ public override void Initialize(AnalysisContext context)
 
                 context.RegisterOperationBlockAction(context =>
                 {
-                    foreach (var operationBlock in context.OperationBlocks)
+                    try
                     {
-                        TrimDataFlowAnalysis trimDataFlowAnalysis = new(context, dataFlowAnalyzerContext, operationBlock);
-                        trimDataFlowAnalysis.InterproceduralAnalyze();
-                        trimDataFlowAnalysis.ReportDiagnostics(context.ReportDiagnostic);
+                        foreach (var operationBlock in context.OperationBlocks)
+                        {
+                            TrimDataFlowAnalysis trimDataFlowAnalysis = new(context, dataFlowAnalyzerContext, operationBlock);
+                            trimDataFlowAnalysis.InterproceduralAnalyze();
+                            trimDataFlowAnalysis.ReportDiagnostics(context.ReportDiagnostic);
+                        }
+                    }
+                    catch (InvalidCastException)
+                    {
+                        // Newer language features may produce NonErrorNamedTypeSymbol
+                        // which will fail to cast to a given target type. Catch and
+                        // ignore the failure in this scenario since it otherwise blocks
+                        // the analyzer
                     }
```

Comment:
@dotnet/illink this was being hit for the new user-defined compound assignment operators.

I wasn't able to get a stack trace or figure out how to get this to debug, so I couldn't determine what wasn't handling the scenario correctly. But something in the `InterproceduralAnalyze()` call gets a `NonErrorNamedTypeSymbol` and then tries to cast it to an `IMethodSymbol`, which fails and brings down the whole analyzer.

Reply:
I think this might be from one of the `(IMethodSymbol)parameter.ContainingSymbol` casts; since `extension(T name) { }` is now possible and *not* a method.